src/plugins/noPendingCount.ts

```diff
@@ -0,0 +1,97 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByPropsLazy } from "@webpack";
+import { definePluginSettings } from "@api/Settings";
+
+const MessageRequestStore = findByPropsLazy("getMessageRequestsCount");
+
+const settings = definePluginSettings({
+    hideFriendRequestsCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide incoming friend requests count",
+        default: true,
+        restartNeeded: true
+    },
+    hideMessageRequestsCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide message requests count",
+        default: true,
+        restartNeeded: true
+    },
+    hidePremiumOffersCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide nitro offers count",
+        default: true,
+        restartNeeded: true
+    }
+});
+
+export default definePlugin({
+    name: "NoPendingCount",
+    description: "Removes the ping count of incoming friend requests, message requests, and nitro offers.",
+    authors: [Devs.amia],
+
+    settings: settings,
+
+    // Functions used to determine the top left count indicator can be found in the single module that calls getUnacknowledgedOffers(...)
+    // or by searching for "showProgressBadge:"
+    patches: [
+        {
+            find: ".getPendingCount=",
+            predicate: () => settings.store.hideFriendRequestsCount,
+            replacement: {
+                match: /(\i)\.getPendingCount=function\(\){return\s\i};/,
+                replace: "$1.getPendingCount=function(){return 0};"
+            }
+        },
+        {
+            find: ".getMessageRequestsCount=",
+            predicate: () => settings.store.hideMessageRequestsCount,
+            replacement: {
+                match: /(\i)\.getMessageRequestsCount=function\(\){return \i\.size};/,
+                replace: "$1.getMessageRequestsCount=function(){return 0};"
+            }
+        },
+        // This prevents the Message Requests tab from always hiding due to the previous patch (and is compatible with spam requests)
+        // In short, only the red badge is hidden. Button visibility behavior isn't changed.
+        {
+            find: ".getSpamChannelsCount(),",
+            predicate: () => settings.store.hideMessageRequestsCount,
+            replacement: {
+                match: /(,\i=\i\.getSpamChannelsCount\(\)),(\i)=\i\.getMessageRequestsCount\(\),(\i=\i>0\|\|\i>0;return\s*?\i}})/,
```

Comment:
```suggestion
                match: /(?<=getSpamChannelsCount\(\),\i=)\i\.getMessageRequestsCount\(\)/,
```

Now your replacement is just `$self.getRealMessageRequestCount()`